w3w-autosuggest/admin/class-w3w-autosuggest-admin.php

```diff
@@ -192,137 +202,162 @@ public function plugin_row_meta( $links, $file ) {
+      $settings = get_option($this->settings_name);
+      $words = get_post_meta($order_id, '_shipping_w3w', true);
+      $nearest_place = get_post_meta($order_id, '_shipping_nearest_place', true);
+      $lat = get_post_meta($order_id, '_shipping_w3w_lat', true);
+      $lng = get_post_meta($order_id, '_shipping_w3w_lng', true);
       $label = $settings['enable_label'] ? $settings['label'] : 'w3w Address';
 
-      require plugin_dir_path( __FILE__ ) . 'partials/add-address-to-order-page.php';
+      require plugin_dir_path(__FILE__) . 'partials/add-address-to-order-page.php';
 
     }
-    
+
     /**
      * Add the 3wa to the customer details for the order email
      * 
      * @since 4.0.6
      */
-    public function add_w3w_address_to_email( $array, $sent_to_admin, $order ) {
+    public function add_w3w_address_to_email($array, $sent_to_admin, $order)
+    {
 
       $order_id = $order->get_id();
-      $settings = get_option( $this->settings_name );
+      $settings = get_option($this->settings_name);
       $label = $settings['enable_label'] ? $settings['label'] : 'w3w Address';
-      $words = get_post_meta( $order_id, '_shipping_w3w', true );
+      $words = get_post_meta($order_id, '_shipping_w3w', true);
 
       $array = [
-        [ 'label' => $label, 'value' => $words ],
+        ['label' => $label, 'value' => $words],
       ];
 
       return $array;
 
     }
 
-    public function set_order_data( $data, $order ) {
-      
+    public function set_order_data($data, $order)
+    {
       $order_id = $order->get_id();
-      $settings = get_option( $this->settings_name );
-      $words = get_post_meta( $order_id, '_shipping_w3w', true );
+      $settings = get_option($this->settings_name);
+      $words = get_post_meta($order_id, '_shipping_w3w', true);
 
-      if ( !!$words ) {
-        
+      if (!!$words) {
         $data['formatted_shipping_address'] .= '<br/>' . $words;
         $data['shipping_address_map_url'] = 'https://what3words.com/' .
-                                            str_replace( '///', '', $words ) .
-                                            '?application=wordpress';
-      
+          str_replace('///', '', $words) .
+          '?application=wordpress';
       }
 
       return $data;
 
     }
 
-    public function run() {
+    public function run()
+    {
 
-      require_once plugin_dir_path( __FILE__ ) . 'partials/w3w-autosuggest-admin-display.php';
+      require_once plugin_dir_path(__FILE__) . 'partials/w3w-autosuggest-admin-display.php';
 
     }
 
-    private function handle_form_submission() {
+    private function handle_form_submission($post)
+    {
+
+      $settings = get_option($this->settings_name);
 
-      $settings = get_option( $this->settings_name );
+      foreach ($post as $key => $val) {
+        $post[$key] = $key === 'clip_to_polygon' ? esc_attr(sanitize_textarea_field($val)) : esc_attr(sanitize_text_field($val));
```

Comment:
Is there any way to check if the value is from a textarea, so we don't need knowledge of any field here?

Reply:
Fair enough, let's hope we remember this is here if we add any other textarea fields 😅